zerver/lib/default_streams.py

```diff
@@ -5,21 +5,29 @@
 
 
 def get_default_streams_for_realm(realm_id: int) -> List[Stream]:
+    # TODO: Deprecate this extremely expensive query. We can't immediately
+    #       just improve it by removing select_related(), because then you
+    #       may get the opposite problem of multiple round trips.
```

Comment:
I think we probably will have a handful of similar cleanups we can do by auditing `git grep Stream[.]objects.select_related`.

`git grep '[.]objects.select_related\(\)'` finds 37 total places in the codebase where we do this sort of thing, and probably all of them are bad ideas now that we have e.g. `user_profile.realm.foo_group_id` type fields.

Probably worth a thread in chat,zulip.org about this since it's a pretty valuable find.